taichi/runtime/llvm/llvm_runtime_executor.cpp

```diff
@@ -600,14 +597,16 @@ void LlvmRuntimeExecutor::materialize_runtime(MemoryPool *memory_pool,
 
     CUDADriver::get_instance().memset(preallocated_device_buffer_, 0,
```

Comment:
nit: shall well add a note here to tell readers the first `result_buffer_size` of preallocated memory is always result buffer? 

```
| ==================preallocated device buffer =========================== |
|<-reserved for return ->|                        
```
or a prettier one :P ?

Reply:
This seems quite important tbh. I guess a better option is to store two pointers one with this offset baked in for the device to suballocate runtime memory?

Reply:
Do we need to keep the original pointer? IIUC this pointer is passed into the runtime allocator and never used again on the host.

Reply:
I will add the comment in another PR because the comment will change in that PR.